internal/collections/syncmap.go

```diff
@@ -13,7 +13,7 @@ type SyncMap[K comparable, V any] struct {
 
 func (s *SyncMap[K, V]) Load(key K) (value V, ok bool) {
 	val, ok := s.m.Load(key)
-	if !ok {
+	if !ok || val == nil {
```

Comment:
I don't understand when this would happen. If the map holds `V`, why would we ever have stored a `nil` except when `V` is `any` and therefore the conversion should succeed?